### DISCUSSION TITLE
Overlay State not replicating to Autonomous Proxy

### DISCUSSION THREAD

**Discussion by AndrewCharlesElliott** [Sixze/ALS-Refactored#219] [Q&A]
Hi I was wondering if anyone could help. I’ve created a weapon class and basic pickup system in C++ so I am not using the normal ALS HUD overlay switcher. When I equip a pickup I’m calling SetOverlayMode passing in the new overlay gameplay tag and it is replicating the new overlay state to the server and other players but not to the local client? Any ideas what I’m missing? Any pointers would be greatly appreciated!

**Comment by ameaninglessname** [comment]
OverlayMode is set to skip owner for replication.
`SetOverlayMode` should be called on the autonomous proxy.

Did you call it on server?

**Reply by ShadowGamingDev** [reply]
Thanks for getting back to me I really appreciate it! So i'm fairly new to multiplayer programming so please bear with me, i'm used to making single player games! I'll try and add more detail as to what i've missed as it'll probably be something obvious to coders with replication experience!

So i've created a Weapon class and added a private  member variable which is an FGameplayTag 'Weapon Overlay' with a public getter. I also have a combat component which I have added to my Character which is a child of the AlsCharacter. So when overlapping a weapon the player can press the 'Equip' Button. If HasAuthority() is true this fires the EquipWeapon() on the Combat Component, if not, ServerEquipWeapon() is called which calls the same EquipWeapon() but on the Server Reliably. 

On the Combat Component EquipWeapon() I attach the weapon to the Character  and then I used the code:

Character->SetOverlayMode(EquippedWeapon->GetWeaponOverlay());

So the Character is my Child Character variable. SetOverlayMode() is the original unchanged public function on the AlsCharacter class, EquippedWeapon is the weapon being equipped and GetWeaponOverlay() is my public getter from the weapon class to get the Overlay FGameplayTag.

Hopefully you'll be able to spot where I've gone wrong although I've just seen the condition on the AlsCharacter that is SkipOwner for the replication! Thanks for pointing that out!

**Reply by ameaninglessname** [reply]
> Character->SetOverlayMode(EquippedWeapon->GetWeaponOverlay());

Just make sure it get called on autonomous proxy and [finish reading "Network Multiplayer Programming" section](https://docs.unrealengine.com/5.1/en-US/networking-and-multiplayer-in-unreal-engine/)

**Reply by ShadowGamingDev** [reply]
All sorted. Thanks for your help!
